src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlCommand.cs

```diff
@@ -934,6 +947,12 @@ override public object ExecuteScalar()
                 statistics = SqlStatistics.StartTimer(Statistics);
                 SqlDataReader ds;
                 ds = RunExecuteReader(0, RunBehavior.ReturnImmediately, returnStream: true);
```

Comment:
I think you meant to replace this line with the if/else. Otherwise it gets executed twice.